pkg/dss/auth/auth.go

```diff
@@ -140,19 +139,18 @@ func (r *JWKSResolver) ResolveKey(ctx context.Context) (interface{}, error) {
 
 // Authorizer authorizes incoming requests.
 type Authorizer struct {
-	logger           *zap.Logger
-	keyGuard         sync.RWMutex
-	key              interface{}
-	requiredScopes   map[string][]string
-	requiredAudience string
+	logger            *zap.Logger
+	key               interface{}
+	requiredScopes    map[string][]string
+	requiredAudiences map[string]bool
 }
 
 // Configuration bundles up creation-time parameters for an Authorizer instance.
 type Configuration struct {
 	KeyResolver       KeyResolver         // Used to initialize and periodically refresh keys.
 	KeyRefreshTimeout time.Duration       // Keys are refreshed on this cadence.
 	RequiredScopes    map[string][]string // RequiredScopes are enforced if not nil.
-	RequiredAudience  string              // RequiredAudience is enforced if not empty.
+	RequiredAudiences []string            // RequiredAudiences is enforced if not empty.
```

Comment:
Enforced always, right?  It seems like we get an error if `acceptable_jwt_audiences` isn't specified